### PR TITLE
Install UKI and sd-boot binaries to ESP

### PR DESCRIPTION
**Pull request by trungams**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./LICENSES-AND-NOTICES/SPECS/data/licenses.json`, `./LICENSES-AND-NOTICES/SPECS/LICENSES-MAP.md`, `./LICENSES-AND-NOTICES/SPECS/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Install UKI and sd-boot binaries to ESP from spec file without having to use post install script. This ensures the EFI binaries under /efi will be properly owned by each package (kernel-uki, systemd-boot)

###### Change Log  

- uki: Install UKI to `/efi/EFI/Linux/vmlinuz-uki-%{kernelver}.efi`
- systemd-boot: Install sd-boot binary to `/efi/EFI/BOOT/grubx64.efi` (might change when we upgrade shim to 15.8)

###### Does this affect the toolchain?  

**YES**

###### Test Methodology

- Local package and image build. New CVM image boots and binaries under ESP are owned correctly by kernel-uki and systemd-boot
- grub2-efi-binary fails to install when systemd-boot is available - this is expected

### CODE REVIEW

**Review by ms-mahuber — APPROVED** [review]

File: SPECS/kernel/kernel-uki.spec [hunk 1]
```diff
@@ -65,11 +65,19 @@ install -vdm 700 %{buildroot}/lib/modules/%{kernelver}
 install -vm 600 vmlinuz-uki.efi %{buildroot}/boot/vmlinuz-uki-%{kernelver}.efi
 ln -s /boot/vmlinuz-uki-%{kernelver}.efi %{buildroot}/lib/modules/%{kernelver}/vmlinuz-uki.efi
 
+# install to ESP
+install -vdm 700 %{buildroot}/boot/efi/EFI/Linux
+cp %{buildroot}/boot/vmlinuz-uki-%{kernelver}.efi %{buildroot}/boot/efi/EFI/Linux/vmlinuz-uki-%{kernelver}.efi
```
Comment (PawelWMS) [inline SPECS/kernel/kernel-uki.spec:70]: Why copy instead of a symbolic link like what's happening a few lines above?

Comment (trungams) [reply SPECS/kernel/kernel-uki.spec:70]: The ESP is a FAT32 partition, so symlinks are not supported

File: SPECS/systemd/systemd.spec [hunk 2]
```diff
@@ -890,6 +890,11 @@ ln -s --relative %{buildroot}%{_bindir}/kernel-install %{buildroot}%{_sbindir}/i
 # Split files in build root into rpms
 python3 %{SOURCE2} %buildroot %{!?want_bootloader:--no-bootloader}
 
+%if 0%{?want_bootloader}
+mkdir -p %{buildroot}/boot/efi/EFI/BOOT
+cp %{buildroot}/usr/lib/systemd/boot/efi/systemd-bootx64.efi %{buildroot}/boot/efi/EFI/BOOT/grubx64.efi
```
Comment (PawelWMS) [inline SPECS/systemd/systemd.spec:895]: Similar question here about copy vs linking. Linking will minimize package size bloat.

Comment (Camelron) [inline SPECS/systemd/systemd.spec:895]: Where does the requirement come from for sd-boot to masquerade as grub?

**Review by Camelron — APPROVED** [review]

File: SPECS/systemd/systemd.spec (hunk 2 above)
Comment (trungams) [reply SPECS/systemd/systemd.spec:895]: Same reasoning as above - ESP (FAT32) does not support symlink

Comment (trungams) [reply SPECS/systemd/systemd.spec:895]: default loader is hardcoded to be grubx64.efi on x86-64

Comment (trungams) [reply SPECS/systemd/systemd.spec:895]: https://github.com/rhboot/shim/blob/shim-15.3/shim.h#L69

File: SPECS/kernel/kernel-uki.spec (hunk 1 above)
Comment (PawelWMS) [reply SPECS/kernel/kernel-uki.spec:70]: Thank you, I've learned something!:)

**Review by PawelWMS — APPROVED** [review]

File: SPECS/kernel/kernel-uki.spec [hunk 3]
```diff
@@ -65,11 +65,19 @@ install -vdm 700 %{buildroot}/lib/modules/%{kernelver}
...
+
 %files
 /boot/vmlinuz-uki-%{kernelver}.efi
 /lib/modules/%{kernelver}/vmlinuz-uki.efi
+/boot/efi/EFI/Linux/vmlinuz-uki-%{kernelver}.efi
```
Comment (christopherco) [inline SPECS/kernel/kernel-uki.spec:75]: Is there a particular reason why we chose "EFI/Linux" for the directory name? Is this naming choice part of a standard?

File: SPECS/systemd/systemd.spec [hunk 4]
```diff
@@ -1174,6 +1179,7 @@ fi
 %if 0%{?want_bootloader}
 %files ukify -f .file-list-ukify
 %files boot -f .file-list-boot
+/boot/efi/EFI/BOOT/grubx64.efi
```
Comment (christopherco) [inline SPECS/systemd/systemd.spec:1182]: This is the same location that grub will install its efi binary in. Likely you will need a package Conflicts here on the grub-efi package

**Review by christopherco — CHANGES_REQUESTED** [review]

File: SPECS/kernel/kernel-uki.spec (hunk 3 above)
Comment (trungams) [reply SPECS/kernel/kernel-uki.spec:75]: That's the preferred location to store UKIs according to the [Boot loader specification](https://uapi-group.org/specifications/specs/boot_loader_specification/#type-2-efi-unified-kernel-images). systemd-boot follows this spec to search for UKIs under $BOOT/EFI/Linux

File: SPECS/systemd/systemd.spec (hunk 4 above)
Comment (trungams) [reply SPECS/systemd/systemd.spec:1183]: Updated

File: SPECS/systemd/systemd.spec [hunk 5]
```diff
@@ -507,6 +507,7 @@ Requires(postun): systemd
 # obsolete parent package so that dnf will install new subpackage on upgrade (#1260394)
 Obsoletes:      %{name} < 229-5
 # Bias the system towards libcurl-minimal if nothing pulls in full libcurl (#1997040)
+Conflicts:      grub2-efi-binary
```
Comment (christopherco) [inline SPECS/systemd/systemd.spec:510]: I think this "Conflicts" should be in the %package boot section, right?

**Review by christopherco — CHANGES_REQUESTED** [review]

File: SPECS/systemd/systemd.spec (hunk 5 above)
Comment (trungams) [reply SPECS/systemd/systemd.spec]: Sorry. That was my 2am brain

**Review by christopherco — APPROVED** [review]

**Comment by trungams** [comment]
:shipit:

**Pull request merged**